File: java/sqrtx.java
public class Solution {
    public int sqrt(int x) {
        // Start typing your Java solution below
        // DO NOT write main() function
        double t = x;
        while(Math.abs(t*t-x)>1e-6){
            t = (t + x/t)/2;
        }
        return (int)t;
    }
}

File: cpp/Sqrtx.cpp
class Solution {
public:
    int sqrt(int x) {
        double t = 1.0;
        while(true) {
            double t1 = 0.5*(t+x/t);
            if(fabs(t-t1)<1e-6) {
                break;
            }
            t = t1;
        }
        return int(t);
    }
};
